packages/recoil/core/Recoil_RecoilEnv.js

```diff
@@ -12,14 +12,61 @@
 'use strict';
 
 const err = require('recoil-shared/util/Recoil_err');
+const gkx = require('recoil-shared/util/Recoil_gkx');
 
 export type RecoilEnv = {
   RECOIL_DUPLICATE_ATOM_KEY_CHECKING_ENABLED: boolean,
+  RECOIL_GKS_ENABLED_UNSTABLE: string[],
 };
 
-const env: RecoilEnv = {
-  RECOIL_DUPLICATE_ATOM_KEY_CHECKING_ENABLED: true,
-};
+const env: RecoilEnv = new Proxy(
+  {
+    RECOIL_DUPLICATE_ATOM_KEY_CHECKING_ENABLED: true,
+    RECOIL_GKS_ENABLED_UNSTABLE: [],
+  },
+  {
+    set(obj, prop, value) {
+      if (prop === 'RECOIL_GKS_ENABLED_UNSTABLE') {
+        // Remove old GKs set in the environment, filter out any GKS that are
+        // already set by default, and set new GKs.
+        obj[prop].forEach(gkx.setFail);
+        obj[prop] = value.filter(gk => !gkx(gk));
```

Comment:
I'm a little confused why it's necessary to call `setFail()` for all the enabled GKs here?  Wouldn't the following `gkx()` call then always fail?

Seems the default GKs should also be added to this environment object for introspection.

Reply:
This is basically to work around the default GKs not being present, so if we move it around so that it's in the shared package I don't think it would be necessary.

Reply:
ok, so moving it really sound a lot cleaner then!